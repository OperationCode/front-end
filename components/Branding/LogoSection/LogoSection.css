.LogoSection ul {
  margin-bottom: 1.5rem;
}

.logoSizeList {
  margin: 0;
  padding: 0;
  list-style: none;
  width: 100%;
}

.logoSizeListItem {
  margin: 0;
  padding: 0;
  width: 100%;
}

.badgeList {
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
  border-bottom: 1px solid #bbbbbb;
}

.largeLogos .badgeList {
  flex-wrap: nowrap;
  flex-direction: column;
}

.largeLogos .badgeList img {
  height: 100%;
}

.badgeList span {
  font-size: 1.5rem;
  font-weight: bold;
}
