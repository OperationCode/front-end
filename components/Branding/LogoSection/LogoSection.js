import React from 'react';
import classNames from 'classnames';
import { s3 } from 'common/constants/urls';
import Section from 'components/_common_/Section/Section';
import Badge from 'components/Badge/Badge';
import styles from './LogoSection.css';

function LogoSection() {
  return (
    <Section contentClassName={styles.LogoSection} title="Logo" theme="gray">
      <ul className={styles.logoSizeList}>
        {/* - SMALL LOGOS - */}
        <li className={classNames(styles.logoSizeListItem, styles.smallLogos)}>
          <h5>SMALL</h5>
          <p>For use when the medal is between 0 and 1-inch tall.</p>

          <ul className={styles.logoTypeList}>
            <li className={styles.logoTypeListItem}>
              <h6>TYPICAL</h6>

              <ul className={styles.badgeList}>
                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/small-blue-logo.png`}
                        alt="Small Blue Accented Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Blue"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/small-red-logo.png`}
                        alt="Small Red Accented Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Red"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img src={`${s3}branding/logos/small-logo.png`} alt="Small Unaccented Logo" />
                    }
                    isImageFirst={false}
                    label="Slate"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/small-white-logo.png`}
                        alt="Small White Accented Logo"
                      />
                    }
                    isImageFirst={false}
                    label="White"
                  />
                </li>
              </ul>
            </li>

            <li className={styles.logoTypeListItem}>
              <h6>STACKED</h6>

              <ul className={styles.badgeList}>
                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/small-stacked-logo-blue.png`}
                        alt="Stacked Small Blue Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Blue"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/small-stacked-logo-red.png`}
                        alt="Stacked Small Red Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Red"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/small-stacked-logo.png`}
                        alt="Stacked Unaccented Slate Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Slate"
                  />
                </li>
              </ul>
            </li>

            <li className={styles.logoTypeListItem}>
              <h6>MEDALS</h6>
              <p>
                For use as an icon reference, when no other logo representations will fit the media
                space.
              </p>

              <ul className={styles.badgeList}>
                <li>
                  <Badge
                    icon={<img src={`${s3}branding/logos/large-blue-medal.png`} alt="Blue Medal" />}
                    isImageFirst={false}
                    label="Blue"
                  />
                </li>

                <li>
                  <Badge
                    icon={<img src={`${s3}branding/logos/large-red-medal.png`} alt="Red Medal" />}
                    isImageFirst={false}
                    label="Red"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img src={`${s3}branding/logos/large-slate-medal.png`} alt="Slate Medal" />
                    }
                    isImageFirst={false}
                    label="Slate"
                  />
                </li>
              </ul>
            </li>
          </ul>
        </li>

        {/* - LARGE LOGOS - */}
        <li className={classNames(styles.logoSizeListItem, styles.largeLogos)}>
          <h5>LARGE</h5>
          <p>For use when the medal is above 1-inch tall.</p>

          <ul className={styles.logoTypeList}>
            <li className={styles.logoTypeListItem}>
              <h6>TYPICAL</h6>

              <ul className={styles.badgeList}>
                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/large-blue-logo.png`}
                        alt="Large Blue Accented Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Blue"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/large-red-logo.png`}
                        alt="Large Red Accented Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Red"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/large-logo.png`}
                        alt="Large Unaccenteded Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Slate"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/large-white-logo.png`}
                        alt="Large White Accented Logo"
                      />
                    }
                    isImageFirst={false}
                    label="White"
                  />
                </li>
              </ul>
            </li>

            <li className={styles.logoTypeListItem}>
              <h6>STACKED</h6>

              <ul className={styles.badgeList}>
                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/large-stacked-logo-blue.png`}
                        alt="Stacked Large Blue Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Blue"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/large-stacked-logo-red.png`}
                        alt="Stacked Large Red Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Red"
                  />
                </li>

                <li>
                  <Badge
                    icon={
                      <img
                        src={`${s3}branding/logos/large-stacked-logo.png`}
                        alt="Stacked Large Slate Logo"
                      />
                    }
                    isImageFirst={false}
                    label="Slate"
                  />
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </Section>
  );
}
export default LogoSection;
