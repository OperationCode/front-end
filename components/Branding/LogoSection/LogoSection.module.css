.logoSizeList {
  margin: 0;
  padding: 0;
  list-style: none;
  width: 100%;
}

.logoSizeList img {
  height: auto;
  max-width: 100%;
}

.logoSizeList ul {
  margin-bottom: 1.5rem;
}

.logoSizeListItem {
  margin: 0;
  padding: 0;
  width: 100%;
}

.badgeList {
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
  border-bottom: 1px solid #bbbbbb;
}

.largeLogos .badgeList {
  flex-wrap: nowrap;
  flex-direction: column;
}

.badgeList span {
  font-size: 1.5rem;
  font-weight: bold;
}
