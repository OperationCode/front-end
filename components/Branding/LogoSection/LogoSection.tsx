import classNames from 'classnames';
import Badge from 'components/Badge/Badge';
import Content from 'components/Content/Content';
import { s3 } from 'common/constants/urls';
import Image from 'next/image';
import styles from './LogoSection.module.css';

function LogoSection() {
  return (
    <Content
      title="Logo"
      theme="gray"
      hasTitleUnderline
      columns={[
        <ul className={styles.logoSizeList}>
          {/* - SMALL LOGOS - */}
          <li className={classNames(styles.logoSizeListItem, styles.smallLogos)}>
            <h5>Small Logos</h5>
            <p>
              For use when Operation Code&apos;s logo name is between 0-1 inch in height. In most
              cases, use the Original Small Logo. The Stacked Small Logo is to be used where
              graphics needs are larger in vertical height than horizontal width with the Operation
              Code logo name still under 1 inch in height.
            </p>

            <ul className={styles.logoTypeList}>
              <li className={styles.logoTypeListItem}>
                <h6>TYPICAL</h6>
                <ul className={styles.badgeList}>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/small-blue-logo.png`}
                          alt="Small Blue Accented Logo"
                          width={318}
                          height={60}
                        />
                      }
                      isImageFirst={false}
                      label="Blue"
                    />
                  </li>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/small-red-logo.png`}
                          alt="Small Red Accented Logo"
                          width={318}
                          height={60}
                        />
                      }
                      isImageFirst={false}
                      label="Red"
                    />
                  </li>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/small-logo.png`}
                          alt="Small Unaccented Logo"
                          width={318}
                          height={60}
                        />
                      }
                      isImageFirst={false}
                      label="Slate"
                    />
                  </li>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/small-white-logo.png`}
                          alt="Small White Accented Logo"
                          width={318}
                          height={60}
                        />
                      }
                      isImageFirst={false}
                      label="White"
                    />
                  </li>
                </ul>
              </li>

              <li className={styles.logoTypeListItem}>
                <h6>Stacked Logos</h6>
                <p>
                  Use the Large Stacked Original logo in areas where you have more vertical space
                  than horizontal. Only use the Red Stacked Logo for special holidays such as
                  Thanksgiving, fall events, and Christmas. The Light Stacked Logo is to be used
                  with dark or contrasting bright colored backgrounds.
                </p>
                <ul className={styles.badgeList}>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/small-stacked-logo-blue.png`}
                          alt="Large Stacked Original"
                          width={270}
                          height={226}
                        />
                      }
                      isImageFirst={false}
                      label="Blue"
                    />
                  </li>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/small-stacked-logo-red.png`}
                          alt="Red Stacked Logo"
                          width={270}
                          height={226}
                        />
                      }
                      isImageFirst={false}
                      label="Red"
                    />
                  </li>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/small-stacked-logo.png`}
                          alt="Light Stacked Logo"
                          width={270}
                          height={226}
                        />
                      }
                      isImageFirst={false}
                      label="Slate"
                    />
                  </li>
                </ul>
              </li>

              <li className={styles.logoTypeListItem}>
                <h6>Medals</h6>
                <p>
                  Use the following medals when you are prompted to upload a thumbnail logo only or
                  profile photo. In most cases, use the OC Blue Medal. The Red Medal should only be
                  used for holidays and special events (such as Thanksgiving, fall, or Christmas).
                  The Navy Medal should be used memorandums or lighter bright colored backgrounds.
                </p>

                <ul className={styles.badgeList}>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-blue-medal.png`}
                          alt="OC Blue Medal"
                          width={300}
                          height={404}
                        />
                      }
                      isImageFirst={false}
                      label="Blue"
                    />
                  </li>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-red-medal.png`}
                          alt="Red Medal"
                          width={300}
                          height={404}
                        />
                      }
                      isImageFirst={false}
                      label="Red"
                    />
                  </li>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-slate-medal.png`}
                          alt="Navy Medal"
                          width={300}
                          height={404}
                        />
                      }
                      isImageFirst={false}
                      label="Navy"
                    />
                  </li>
                </ul>
              </li>

              <li className={styles.logoTypeListItem}>
                <h6>Badges</h6>
                <p>
                  You may also use badges as a profile photo or thumbnail. Badges should be used to
                  accompany corporate sponsors or alliance partnerships on shared branding. Use the
                  original OC Blue Badge in most cases, except the Red Badge for special events and
                  holidays (Thanksgiving, fall and Christmas). The Navy Badge may be used with light
                  backgrounds and bright colored backgrounds.
                </p>
                <ul className={styles.badgeList}>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-blue-medal.png`}
                          alt="OC Blue Badge"
                          width={300}
                          height={404}
                        />
                      }
                      isImageFirst={false}
                      label="OC Blue Badge"
                    />
                  </li>

                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-red-medal.png`}
                          alt="Red Badge"
                          width={300}
                          height={404}
                        />
                      }
                      isImageFirst={false}
                      label="Red Badge"
                    />
                  </li>

                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-slate-medal.png`}
                          alt="Navy"
                          width={300}
                          height={404}
                        />
                      }
                      isImageFirst={false}
                      label="Navy Badge"
                    />
                  </li>
                </ul>
              </li>
            </ul>
          </li>

          {/* - LARGE LOGOS - */}
          <li className={classNames(styles.logoSizeListItem, styles.largeLogos)}>
            <h5>Large Logos</h5>
            <p>
              Use the Original Large Logo in almost all cases above 1 inch in height. The Red Large
              Logo is to be used in special events (such as Thanksgiving, fall or Christmas), and
              the Light Large Logo is to be used with dark backgrounds or bright colored
              backgrounds.
            </p>

            <ul className={styles.logoTypeList}>
              <li className={styles.logoTypeListItem}>
                <h6>TYPICAL</h6>

                <ul className={styles.badgeList}>
                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-blue-logo.png`}
                          alt="Original Large Logo"
                          width={384}
                          height={70}
                        />
                      }
                      isImageFirst={false}
                      label="Blue"
                    />
                  </li>

                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-red-logo.png`}
                          alt="Red Large Logo"
                          width={384}
                          height={70}
                        />
                      }
                      isImageFirst={false}
                      label="Red"
                    />
                  </li>

                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-logo.png`}
                          alt="Large Unaccented Logo"
                          width={384}
                          height={70}
                        />
                      }
                      isImageFirst={false}
                      label="Slate"
                    />
                  </li>

                  <li>
                    <Badge
                      icon={
                        <Image
                          src={`${s3}branding/logos/large-white-logo.png`}
                          alt="Large White Accented Logo"
                          width={384}
                          height={70}
                        />
                      }
                      isImageFirst={false}
                      label="White"
                    />
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>,
      ]}
    />
  );
}
export default LogoSection;
