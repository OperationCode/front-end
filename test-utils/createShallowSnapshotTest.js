import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
