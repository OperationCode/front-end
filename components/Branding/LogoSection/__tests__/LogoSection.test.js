import React from 'react';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import LogoSection from '../LogoSection';

describe('LogoSection', () => {
  it('should render with required props', () => {
    createShallowSnapshotTest(<LogoSection />);
  });
});
