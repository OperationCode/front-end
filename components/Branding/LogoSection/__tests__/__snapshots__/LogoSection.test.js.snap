// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoSection should render with required props 1`] = `
<Section
  contentClassName="LogoSection"
  hasHeadingLines={true}
  id=""
  style={Object {}}
  theme="gray"
  title="Logo"
>
  <ul
    className="logoSizeList"
  >
    <li
      className="logoSizeListItem smallLogos"
    >
      <h5>
        SMALL
      </h5>
      <p>
        For use when the medal is between 0 and 1-inch tall.
      </p>
      <ul
        className="logoTypeList"
      >
        <li
          className="logoTypeListItem"
        >
          <h6>
            TYPICAL
          </h6>
          <ul
            className="badgeList"
          >
            <li>
              <Badge
                icon={
                  <img
                    alt="Small Blue Accented Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/small-blue-logo.png"
                  />
                }
                isImageFirst={false}
                label="Blue"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Small Red Accented Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/small-red-logo.png"
                  />
                }
                isImageFirst={false}
                label="Red"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Small Unaccented Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/small-logo.png"
                  />
                }
                isImageFirst={false}
                label="Slate"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Small White Accented Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/small-white-logo.png"
                  />
                }
                isImageFirst={false}
                label="White"
              />
            </li>
          </ul>
        </li>
        <li
          className="logoTypeListItem"
        >
          <h6>
            STACKED
          </h6>
          <ul
            className="badgeList"
          >
            <li>
              <Badge
                icon={
                  <img
                    alt="Stacked Small Blue Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/small-stacked-logo-blue.png"
                  />
                }
                isImageFirst={false}
                label="Blue"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Stacked Small Red Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/small-stacked-logo-red.png"
                  />
                }
                isImageFirst={false}
                label="Red"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Stacked Unaccented Slate Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/small-stacked-logo.png"
                  />
                }
                isImageFirst={false}
                label="Slate"
              />
            </li>
          </ul>
        </li>
        <li
          className="logoTypeListItem"
        >
          <h6>
            MEDALS
          </h6>
          <p>
            For use as an icon reference, when no other logo representations will fit the media space.
          </p>
          <ul
            className="badgeList"
          >
            <li>
              <Badge
                icon={
                  <img
                    alt="Blue Medal"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-blue-medal.png"
                  />
                }
                isImageFirst={false}
                label="Blue"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Red Medal"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-red-medal.png"
                  />
                }
                isImageFirst={false}
                label="Red"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Slate Medal"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-slate-medal.png"
                  />
                }
                isImageFirst={false}
                label="Slate"
              />
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li
      className="logoSizeListItem largeLogos"
    >
      <h5>
        LARGE
      </h5>
      <p>
        For use when the medal is above 1-inch tall.
      </p>
      <ul
        className="logoTypeList"
      >
        <li
          className="logoTypeListItem"
        >
          <h6>
            TYPICAL
          </h6>
          <ul
            className="badgeList"
          >
            <li>
              <Badge
                icon={
                  <img
                    alt="Large Blue Accented Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-blue-logo.png"
                  />
                }
                isImageFirst={false}
                label="Blue"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Large Red Accented Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-red-logo.png"
                  />
                }
                isImageFirst={false}
                label="Red"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Large Unaccenteded Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-logo.png"
                  />
                }
                isImageFirst={false}
                label="Slate"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Large White Accented Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-white-logo.png"
                  />
                }
                isImageFirst={false}
                label="White"
              />
            </li>
          </ul>
        </li>
        <li
          className="logoTypeListItem"
        >
          <h6>
            STACKED
          </h6>
          <ul
            className="badgeList"
          >
            <li>
              <Badge
                icon={
                  <img
                    alt="Stacked Large Blue Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-stacked-logo-blue.png"
                  />
                }
                isImageFirst={false}
                label="Blue"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Stacked Large Red Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-stacked-logo-red.png"
                  />
                }
                isImageFirst={false}
                label="Red"
              />
            </li>
            <li>
              <Badge
                icon={
                  <img
                    alt="Stacked Large Slate Logo"
                    src="https://s3.amazonaws.com/operationcode-assets/branding/logos/large-stacked-logo.png"
                  />
                }
                isImageFirst={false}
                label="Slate"
              />
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</Section>
`;
